pkg/controller/razeedeployment/razeedeployment_controller.go

```diff
@@ -856,275 +871,310 @@ func (r *ReconcileRazeeDeployment) Reconcile(request reconcile.Request) (reconci
-		return reconcile.Result{RequeueAfter: time.Second * 30}, nil
-		// return reconcile.Result{Requeue: true}, nil
-		// return reconcile.Result{}, nil
+		return reconcile.Result{Requeue: true}, nil
+
 	} else if err != nil {
-		reqLogger.Error(err, "Failed to get Job(s) from Cluster")
+		reqLogger.Error(err, "Failed to get RemoteResourceS3 deployment from Cluster")
 		return reconcile.Result{}, err
 	}
 
-	if err := controllerutil.SetControllerReference(instance, foundJob, r.scheme); err != nil {
+	if err := controllerutil.SetControllerReference(instance, rrs3Deployment, r.scheme); err != nil {
 		reqLogger.Error(err, "Failed to set controller reference")
 		return reconcile.Result{}, err
 	}
 
-	job := r.makeRazeeJob(request, instance)
-	if job.Spec.Template.Spec.Containers[0].Image != foundJob.Spec.Template.Spec.Containers[0].Image ||
-		!utils.Equal(job.Spec.Template.Spec.Containers[0].Args, foundJob.Spec.Template.Spec.Containers[0].Args) ||
-		!utils.Equal(job.Spec.Template.Spec.Containers[0].Command, foundJob.Spec.Template.Spec.Containers[0].Command) {
-		reqLogger.Info("Updating job with new values")
-		foundJob.Spec = job.Spec
-		err := r.client.Delete(context.TODO(), foundJob, client.PropagationPolicy(metav1.DeletePropagationBackground))
+	message := "RemoteResourceS3 deployment install finished"
+	instance.Status.Conditions.SetCondition(status.Condition{
+		Type:    marketplacev1alpha1.ConditionInstalling,
+		Status:  corev1.ConditionTrue,
+		Reason:  marketplacev1alpha1.ReasonRazeeRemoteResourceS3DeploymentStart,
+		Message: message,
+	})
+
+	// watch-keeper deployment
+	watchKeeperDeployment := &appsv1.Deployment{}
+	reqLogger.V(0).Info("Finding watch-keeper deployment")
+	err = r.client.Get(context.TODO(), types.NamespacedName{
+		Name:      utils.RHM_WATCHKEEPER_DEPLOYMENT_NAME,
+		Namespace: request.Namespace,
+	}, watchKeeperDeployment)
+	if errors.IsNotFound(err) {
+		reqLogger.V(0).Info("Creating watch-keeper deployment")
+		watchKeeperDeployment = r.makeWatchKeeperDeployment(instance)
+		err = r.client.Create(context.TODO(), watchKeeperDeployment)
 		if err != nil {
-			reqLogger.Error(err, "Failed to update Job(s) from Cluster")
+			reqLogger.Error(err, "Failed to create watch-keeper deployment on cluster")
 			return reconcile.Result{}, err
 		}
-		return reconcile.Result{Requeue: true}, nil
-	}
+		reqLogger.Info("watch-keeper deployment created successfully")
 
-	if !reflect.DeepEqual(instance.Status.JobState, foundJob.Status) {
-		// Update status and conditions
-		instance.Status.JobState = foundJob.Status
-		for _, jobCondition := range foundJob.Status.Conditions {
-			instance.Status.JobConditions = &jobCondition
-		}
+		message := "watch-keeper install starting"
+		instance.Status.Conditions.SetCondition(status.Condition{
+			Type:    marketplacev1alpha1.ConditionInstalling,
+			Status:  corev1.ConditionTrue,
+			Reason:  marketplacev1alpha1.ReasonWatchKeeperDeploymentInstalled,
```

Comment:
Think this is meant to be `ReasonWatchKeeperDeploymentStart`?
```suggestion
			Reason:  marketplacev1alpha1.ReasonWatchKeeperDeploymentStart,
```